lib/coinbase/payload_signature.rb

```diff
@@ -0,0 +1,179 @@
+          addresses_api.create_payload_signature(
+            wallet_id,
+            address_id,
+            create_payload_signature_request: create_payload_signature_request
+          )
+        end
+
+        new(model)
+      end
+
+      # Enumerates the payload signatures for a given address belonging to a wallet.
+      # The result is an enumerator that lazily fetches from the server, and can be iterated over,
+      # converted an array, etc...
+      # @return [Enumerable<Coinbase::PayloadSignature>] Enumerator that returns payload signatures
+      def list(wallet_id:, address_id:)
+        Coinbase::Pagination.enumerate(
+          ->(page) { fetch_page(wallet_id, address_id, page) }
+        ) do |payload_signature|
+          new(payload_signature)
+        end
+      end
+
+      private
+
+      def addresses_api
+        Coinbase::Client::AddressesApi.new(Coinbase.configuration.api_client)
+      end
+
+      def fetch_page(wallet_id, address_id, page)
+        addresses_api.list_payload_signatures(wallet_id, address_id, { limit: DEFAULT_PAGE_LIMIT, page: page })
+      end
+    end
+
+    # Returns a new PayloadSignature object. Do not use this method directly.
+    # Instead use Coinbase::PayloadSignature.create.
+    # @param model [Coinbase::Client::PayloadSignature] The underlying Payload Signature obejct
+    def initialize(model)
+      raise unless model.is_a?(Coinbase::Client::PayloadSignature)
+
+      @model = model
+    end
+
+    # Returns the Payload Signature ID.
+    # @return [String] The Payload Signature ID
+    def id
+      @model.payload_signature_id
+    end
+
+    # Returns the Wallet ID of the Payload Signature.
+    # @return [String] The Wallet ID
+    def wallet_id
+      @model.wallet_id
+    end
+
+    # Returns the Address ID of the Payload Signature.
+    # @return [String] The Address ID
+    def address_id
+      @model.address_id
+    end
+
+    # Returns the Unsigned Payload of the Payload Signature.
+    # @return [String] The Unsigned Payload
+    def unsigned_payload
+      @model.unsigned_payload
+    end
+
+    # Returns the Signature of the Payload Signature.
+    # @return [String] The Signature
+    def signature
+      @signature ||= @model.signature
+    end
+
+    # Returns the status of the Payload Signature.
+    # @return [Symbol] The status
+    def status
+      @model.status
+    end
+
+    # Returns whether the Payload Signature is in a terminal state.
+    # @return [Boolean] Whether the Payload Signature is in a terminal state
+    def terminal_state?
+      Status::TERMINAL_STATES.include?(status)
+    end
+
+    # # Reload reloads the Payload Signature model with the latest version from the server side.
+    # @return [PayloadSignature] The most recent version of Payload Signature from the server
+    def reload
+      @model = Coinbase.call_api do
+        addresses_api.get_payload_signature(wallet_id, address_id, id)
+      end
+
```

Comment:
This won't update the `signature` when creating a payload signature using a server-signer.

We probably need to ensure we do something like `@signature = @model.signature` after fetching the model.

Or if we're never setting `signature` explicitly client-side (since for dev-managed we just explicitly , lets just not memoize it?

Reply:
updated signature instance method to return `@model.signature` to resolve this